diff -ur libebur128_fork_for_freelcs_3.4/ebur128/CMakeLists.txt libebur128_fork_for_freelcs_3.4-patched/ebur128/CMakeLists.txt
--- libebur128_fork_for_freelcs_3.4/ebur128/CMakeLists.txt	2018-04-25 19:11:47.000000000 +0300
+++ libebur128_fork_for_freelcs_3.4-patched/ebur128/CMakeLists.txt	2018-05-05 18:38:00.184149893 +0300
@@ -1,5 +1,5 @@
 cmake_minimum_required(VERSION 2.8.12 FATAL_ERROR)
-project(libebur128 C)
+project(libebur128-freelcs C)
 
 list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
 set(EXECUTABLE_OUTPUT_PATH ${CMAKE_CURRENT_BINARY_DIR})
diff -ur libebur128_fork_for_freelcs_3.4/ebur128/ebur128/CMakeLists.txt libebur128_fork_for_freelcs_3.4-patched/ebur128/ebur128/CMakeLists.txt
--- libebur128_fork_for_freelcs_3.4/ebur128/ebur128/CMakeLists.txt	2018-04-25 19:11:47.000000000 +0300
+++ libebur128_fork_for_freelcs_3.4-patched/ebur128/ebur128/CMakeLists.txt	2018-05-05 18:51:01.721264943 +0300
@@ -29,7 +29,7 @@
 if(BUILD_STATIC_LIBS)
   add_library(ebur128_static STATIC ebur128.c)
   if(NOT MSVC)
-    set_property(TARGET ebur128_static PROPERTY OUTPUT_NAME ebur128)
+    set_property(TARGET ebur128_static PROPERTY OUTPUT_NAME ebur128-freelcs)
   endif()
 endif()
 
@@ -44,23 +44,23 @@
   set(EBUR128_SHARED_SOURCE ${EBUR128_SHARED_SOURCE} ebur128.def)
 endif()
 
-add_library(ebur128 SHARED ${EBUR128_SHARED_SOURCE})
-set_target_properties(ebur128 PROPERTIES
+add_library(ebur128-freelcs SHARED ${EBUR128_SHARED_SOURCE})
+set_target_properties(ebur128-freelcs PROPERTIES
     SOVERSION ${EBUR128_VERSION_MAJOR}
     VERSION ${EBUR128_VERSION})
 
 find_library(MATH_LIBRARY m)
 if(MATH_LIBRARY)
-  target_link_libraries(ebur128 ${MATH_LIBRARY})
+  target_link_libraries(ebur128-freelcs ${MATH_LIBRARY})
 endif()
 
 set(EBUR128_INCLUDE_DIR ${CMAKE_CURRENT_SOURCE_DIR} CACHE INTERNAL "")
 
 install(FILES ebur128.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 if(BUILD_STATIC_LIBS)
-  install(TARGETS ebur128 ebur128_static DESTINATION ${CMAKE_INSTALL_LIBDIR})
+  install(TARGETS ebur128-freelcs ebur128_static DESTINATION ${CMAKE_INSTALL_LIBDIR})
 else()
-  install(TARGETS ebur128 DESTINATION ${CMAKE_INSTALL_LIBDIR})
+  install(TARGETS ebur128-freelcs DESTINATION ${CMAKE_INSTALL_LIBDIR})
 endif()
 
 #### pkg-config
diff -ur libebur128_fork_for_freelcs_3.4/ebur128/ebur128/ebur128.c libebur128_fork_for_freelcs_3.4-patched/ebur128/ebur128/ebur128.c
--- libebur128_fork_for_freelcs_3.4/ebur128/ebur128/ebur128.c	2018-04-25 19:11:47.000000000 +0300
+++ libebur128_fork_for_freelcs_3.4-patched/ebur128/ebur128/ebur128.c	2018-05-05 19:00:21.910631034 +0300
@@ -258,18 +258,6 @@
   if (!st->d->channel_map) {
     return EBUR128_ERROR_NOMEM;
   }
-  if (st->channels == 4) {
-    st->d->channel_map[0] = EBUR128_LEFT;
-    st->d->channel_map[1] = EBUR128_RIGHT;
-    st->d->channel_map[2] = EBUR128_LEFT_SURROUND;
-    st->d->channel_map[3] = EBUR128_RIGHT_SURROUND;
-  } else if (st->channels == 5) {
-    st->d->channel_map[0] = EBUR128_LEFT;
-    st->d->channel_map[1] = EBUR128_RIGHT;
-    st->d->channel_map[2] = EBUR128_CENTER;
-    st->d->channel_map[3] = EBUR128_LEFT_SURROUND;
-    st->d->channel_map[4] = EBUR128_RIGHT_SURROUND;
-  } else {
     for (i = 0; i < st->channels; ++i) {
       switch (i) {
         case 0:  st->d->channel_map[i] = EBUR128_LEFT;           break;
@@ -281,6 +269,17 @@
         default: st->d->channel_map[i] = EBUR128_UNUSED;         break;
       }
     }
+  if (st->channels == 4) {
+    st->d->channel_map[0] = EBUR128_LEFT;
+    st->d->channel_map[1] = EBUR128_RIGHT;
+    st->d->channel_map[2] = EBUR128_LEFT_SURROUND;
+    st->d->channel_map[3] = EBUR128_RIGHT_SURROUND;
+  } else if (st->channels == 5) {
+    st->d->channel_map[0] = EBUR128_LEFT;
+    st->d->channel_map[1] = EBUR128_RIGHT;
+    st->d->channel_map[2] = EBUR128_CENTER;
+    st->d->channel_map[3] = EBUR128_LEFT_SURROUND;
+    st->d->channel_map[4] = EBUR128_RIGHT_SURROUND;
   }
   return EBUR128_SUCCESS;
 }
diff -ur libebur128_fork_for_freelcs_3.4/ebur128/test/CMakeLists.txt libebur128_fork_for_freelcs_3.4-patched/ebur128/test/CMakeLists.txt
--- libebur128_fork_for_freelcs_3.4/ebur128/test/CMakeLists.txt	2018-04-25 19:11:47.000000000 +0300
+++ libebur128_fork_for_freelcs_3.4-patched/ebur128/test/CMakeLists.txt	2018-05-05 18:38:00.184149893 +0300
@@ -15,6 +15,6 @@
       COMPILE_FLAGS " ${SNDFILE_CFLAGS}")
   set_property(TARGET minimal-example APPEND_STRING PROPERTY
       COMPILE_FLAGS " ${SNDFILE_CFLAGS}")
-  target_link_libraries(r128-test-library ebur128 ${SNDFILE_LIBRARIES})
-  target_link_libraries(minimal-example ebur128 ${SNDFILE_LIBRARIES})
+  target_link_libraries(r128-test-library ebur128-freelcs ${SNDFILE_LIBRARIES})
+  target_link_libraries(minimal-example ebur128-freelcs ${SNDFILE_LIBRARIES})
 endif()
diff -ur libebur128_fork_for_freelcs_3.4/scanner/CMakeLists.txt libebur128_fork_for_freelcs_3.4-patched/scanner/CMakeLists.txt
--- libebur128_fork_for_freelcs_3.4/scanner/CMakeLists.txt	2018-04-25 19:11:10.000000000 +0300
+++ libebur128_fork_for_freelcs_3.4-patched/scanner/CMakeLists.txt	2018-05-05 18:38:00.184149893 +0300
@@ -41,23 +41,23 @@
 
 
   add_library(scanner-lib scanner-scan scanner-dump)
-  target_link_libraries(scanner-lib scanner-common ebur128)
+  target_link_libraries(scanner-lib scanner-common ebur128-freelcs)
 
-  add_executable(loudness scanner)
-  target_link_libraries(loudness
+  add_executable(loudness-freelcs  scanner)
+  target_link_libraries(loudness-freelcs 
         scanner-lib filetree input
         ${GLIB20_LIBRARIES} ${GTHREAD20_LIBRARIES})
 
 
 
   if(SNDFILE_FOUND AND NOT DISABLE_SNDFILE)
-    set_property(TARGET loudness APPEND PROPERTY
+    set_property(TARGET loudness-freelcs  APPEND PROPERTY
           COMPILE_DEFINITIONS "USE_SNDFILE")
   endif()
 
   if(TARGET scanner-tag)
-    target_link_libraries(loudness scanner-tag)
-    set_property(TARGET loudness APPEND PROPERTY
+    target_link_libraries(loudness-freelcs  scanner-tag)
+    set_property(TARGET loudness-freelcs  APPEND PROPERTY
           COMPILE_DEFINITIONS "USE_TAGLIB")
   endif()
 endif()
diff -ur libebur128_fork_for_freelcs_3.4/scanner/inputaudio/ffmpeg/CMakeLists.txt libebur128_fork_for_freelcs_3.4-patched/scanner/inputaudio/ffmpeg/CMakeLists.txt
--- libebur128_fork_for_freelcs_3.4/scanner/inputaudio/ffmpeg/CMakeLists.txt	2018-04-25 19:11:10.000000000 +0300
+++ libebur128_fork_for_freelcs_3.4-patched/scanner/inputaudio/ffmpeg/CMakeLists.txt	2018-05-05 18:38:00.184149893 +0300
@@ -23,9 +23,9 @@
                    ${LIBAVUTIL_LIBRARY_DIRS}
                    ${GMODULE20_LIBRARY_DIRS})
 
-  add_library(input_ffmpeg MODULE input_ffmpeg.c ../input_helper.c)
+  add_library(input_ffmpeg-freelcs  MODULE input_ffmpeg.c ../input_helper.c)
 
-  target_link_libraries(input_ffmpeg
+  target_link_libraries(input_ffmpeg-freelcs 
                         ${LIBAVFORMAT_LIBRARIES}
                         ${LIBAVCODEC_LIBRARIES}
                         ${LIBAVUTIL_LIBRARIES}
@@ -39,12 +39,12 @@
 
   if(INPUT_FFMPEG_CFLAGS)
     to_space_list(INPUT_FFMPEG_CFLAGS)
-    set_target_properties(input_ffmpeg PROPERTIES
+    set_target_properties(input_ffmpeg-freelcs  PROPERTIES
             COMPILE_FLAGS ${INPUT_FFMPEG_CFLAGS})
   endif()
   if(INPUT_FFMPEG_LDFLAGS)
     to_space_list(INPUT_FFMPEG_LDFLAGS)
-    set_target_properties(input_ffmpeg PROPERTIES
+    set_target_properties(input_ffmpeg-freelcs  PROPERTIES
                LINK_FLAGS ${INPUT_FFMPEG_LDFLAGS})
   endif()
 endif()
diff -ur libebur128_fork_for_freelcs_3.4/scanner/inputaudio/ffmpeg/input_ffmpeg.c libebur128_fork_for_freelcs_3.4-patched/scanner/inputaudio/ffmpeg/input_ffmpeg.c
--- libebur128_fork_for_freelcs_3.4/scanner/inputaudio/ffmpeg/input_ffmpeg.c	2018-04-25 19:11:10.000000000 +0300
+++ libebur128_fork_for_freelcs_3.4-patched/scanner/inputaudio/ffmpeg/input_ffmpeg.c	2018-05-05 18:38:00.184149893 +0300
@@ -145,6 +145,7 @@
 }
 
 static int ffmpeg_set_channel_map(struct input_handle* ih, int* st) {
+  return 1;
   if (ih->codec_context->channel_layout) {
     unsigned int channel_map_index = 0;
     int bit_counter = 0;
diff -ur libebur128_fork_for_freelcs_3.4/scanner/inputaudio/gstreamer/CMakeLists.txt libebur128_fork_for_freelcs_3.4-patched/scanner/inputaudio/gstreamer/CMakeLists.txt
--- libebur128_fork_for_freelcs_3.4/scanner/inputaudio/gstreamer/CMakeLists.txt	2018-04-25 19:11:10.000000000 +0300
+++ libebur128_fork_for_freelcs_3.4-patched/scanner/inputaudio/gstreamer/CMakeLists.txt	2018-05-05 18:38:00.184149893 +0300
@@ -17,7 +17,7 @@
    NOT DISABLE_GLIB20 AND NOT DISABLE_GSTREAMER)
   include_directories(${INPUT_INCLUDE_DIR} ${EBUR128_INCLUDE_DIR})
 
-  add_library(input_gstreamer MODULE input_gstreamer.c)
+  add_library(input_gstreamer-freelcs  MODULE input_gstreamer.c)
 
   list(APPEND INPUT_GSTREAMER_CFLAGS  ${GMODULE20_CFLAGS}  ${GSTREAMER_APP_CFLAGS}  ${GSTREAMER_AUDIO_CFLAGS})
   list(APPEND INPUT_GSTREAMER_LDFLAGS ${GMODULE20_LDFLAGS} ${GSTREAMER_APP_LDFLAGS} ${GSTREAMER_AUDIO_LDFLAGS})
@@ -25,7 +25,7 @@
   to_space_list(INPUT_GSTREAMER_CFLAGS)
   to_space_list(INPUT_GSTREAMER_LDFLAGS)
 
-  set_target_properties(input_gstreamer PROPERTIES
+  set_target_properties(input_gstreamer-freelcs  PROPERTIES
           COMPILE_FLAGS ${INPUT_GSTREAMER_CFLAGS})
-  target_link_libraries(input_gstreamer ${INPUT_GSTREAMER_LDFLAGS})
+  target_link_libraries(input_gstreamer-freelcs  ${INPUT_GSTREAMER_LDFLAGS})
 endif()
diff -ur libebur128_fork_for_freelcs_3.4/scanner/inputaudio/gstreamer/input_gstreamer.c libebur128_fork_for_freelcs_3.4-patched/scanner/inputaudio/gstreamer/input_gstreamer.c
--- libebur128_fork_for_freelcs_3.4/scanner/inputaudio/gstreamer/input_gstreamer.c	2018-04-25 19:11:10.000000000 +0300
+++ libebur128_fork_for_freelcs_3.4-patched/scanner/inputaudio/gstreamer/input_gstreamer.c	2018-05-05 18:38:00.187483242 +0300
@@ -263,6 +263,7 @@
 }
 
 static int gstreamer_set_channel_map(struct input_handle* ih, int* st) {
+  return 0;
   gint j;
   for (j = 0; j < ih->n_channels; ++j) {
     switch (ih->channel_positions[j]) {
diff -ur libebur128_fork_for_freelcs_3.4/scanner/inputaudio/input.c libebur128_fork_for_freelcs_3.4-patched/scanner/inputaudio/input.c
--- libebur128_fork_for_freelcs_3.4/scanner/inputaudio/input.c	2018-04-25 19:11:10.000000000 +0300
+++ libebur128_fork_for_freelcs_3.4-patched/scanner/inputaudio/input.c	2018-05-05 18:38:00.187483242 +0300
@@ -6,12 +6,12 @@
 #include <stdio.h>
 
 static const char* plugin_names[] = {
-  "input_sndfile",
-  "input_mpg123",
-  "input_musepack",
-  "input_ffmpeg",
+  "input_sndfile-freelcs",
+  "input_mpg123-freelcs",
+  "input_musepack-freelcs",
+  "input_ffmpeg-freelcs",
 #ifndef GSTREAMER_INPUT_STATIC
-  "input_gstreamer",
+  "input_gstreamer-freelcs",
 #endif
   NULL
 };
diff -ur libebur128_fork_for_freelcs_3.4/scanner/inputaudio/mpcdec/CMakeLists.txt libebur128_fork_for_freelcs_3.4-patched/scanner/inputaudio/mpcdec/CMakeLists.txt
--- libebur128_fork_for_freelcs_3.4/scanner/inputaudio/mpcdec/CMakeLists.txt	2018-04-25 19:11:10.000000000 +0300
+++ libebur128_fork_for_freelcs_3.4-patched/scanner/inputaudio/mpcdec/CMakeLists.txt	2018-05-05 18:38:00.187483242 +0300
@@ -14,9 +14,9 @@
   include_directories(SYSTEM ${MPCDEC_INCLUDE_DIRS})
   include_directories(${INPUT_INCLUDE_DIR} ${EBUR128_INCLUDE_DIR})
 
-  add_library(input_musepack MODULE input_mpcdec.c)
+  add_library(input_musepack-freelcs  MODULE input_mpcdec.c)
 
-  target_link_libraries(input_musepack ${MPCDEC_LIBRARIES})
+  target_link_libraries(input_musepack-freelcs  ${MPCDEC_LIBRARIES})
 
   list(APPEND INPUT_MPCDEC_CFLAGS  ${GMODULE20_CFLAGS})
   list(APPEND INPUT_MPCDEC_LDFLAGS ${GMODULE20_LDFLAGS})
@@ -24,7 +24,7 @@
   to_space_list(INPUT_MPCDEC_CFLAGS)
   to_space_list(INPUT_MPCDEC_LDFLAGS)
 
-  set_target_properties(input_musepack PROPERTIES
+  set_target_properties(input_musepack-freelcs PROPERTIES
           COMPILE_FLAGS ${INPUT_MPCDEC_CFLAGS}
              LINK_FLAGS ${INPUT_MPCDEC_LDFLAGS})
 endif()
diff -ur libebur128_fork_for_freelcs_3.4/scanner/inputaudio/mpg123/CMakeLists.txt libebur128_fork_for_freelcs_3.4-patched/scanner/inputaudio/mpg123/CMakeLists.txt
--- libebur128_fork_for_freelcs_3.4/scanner/inputaudio/mpg123/CMakeLists.txt	2018-04-25 19:11:10.000000000 +0300
+++ libebur128_fork_for_freelcs_3.4-patched/scanner/inputaudio/mpg123/CMakeLists.txt	2018-05-05 18:38:00.187483242 +0300
@@ -14,9 +14,9 @@
   include_directories(SYSTEM ${MPG123_INCLUDE_DIRS})
   include_directories(${INPUT_INCLUDE_DIR} ${EBUR128_INCLUDE_DIR})
 
-  add_library(input_mpg123 MODULE input_mpg123.c)
+  add_library(input_mpg123-freelcs  MODULE input_mpg123.c)
 
-  target_link_libraries(input_mpg123 ${MPG123_LIBRARIES})
+  target_link_libraries(input_mpg123-freelcs  ${MPG123_LIBRARIES})
 
   list(APPEND INPUT_MPG123_CFLAGS  ${GMODULE20_CFLAGS})
   list(APPEND INPUT_MPG123_LDFLAGS ${GMODULE20_LDFLAGS})
@@ -24,7 +24,7 @@
   to_space_list(INPUT_MPG123_CFLAGS)
   to_space_list(INPUT_MPG123_LDFLAGS)
 
-  set_target_properties(input_mpg123 PROPERTIES
+  set_target_properties(input_mpg123-freelcs  PROPERTIES
           COMPILE_FLAGS ${INPUT_MPG123_CFLAGS}
              LINK_FLAGS ${INPUT_MPG123_LDFLAGS})
 endif()
diff -ur libebur128_fork_for_freelcs_3.4/scanner/inputaudio/sndfile/CMakeLists.txt libebur128_fork_for_freelcs_3.4-patched/scanner/inputaudio/sndfile/CMakeLists.txt
--- libebur128_fork_for_freelcs_3.4/scanner/inputaudio/sndfile/CMakeLists.txt	2018-04-25 19:11:10.000000000 +0300
+++ libebur128_fork_for_freelcs_3.4-patched/scanner/inputaudio/sndfile/CMakeLists.txt	2018-05-05 18:38:00.187483242 +0300
@@ -14,9 +14,9 @@
   include_directories(${GMODULE20_INCLUDE_DIRS})
   link_directories(${SNDFILE_LIBRARY_DIRS} ${GMODULE20_LIBRARY_DIRS})
 
-  add_library(input_sndfile MODULE input_sndfile.c ../input_helper.c)
+  add_library(input_sndfile-freelcs  MODULE input_sndfile.c ../input_helper.c)
 
-  target_link_libraries(input_sndfile
+  target_link_libraries(input_sndfile-freelcs 
                         ${SNDFILE_LIBRARIES} ${GMODULE20_LIBRARIES})
 
   list(APPEND INPUT_SNDFILE_CFLAGS  ${GMODULE20_CFLAGS_OTHER})
@@ -24,12 +24,12 @@
 
   if(INPUT_SNDFILES_CFLAGS)
     to_space_list(INPUT_SNDFILE_CFLAGS)
-    set_target_properties(input_sndfile PROPERTIES
+    set_target_properties(input_sndfile-freelcs  PROPERTIES
             COMPILE_FLAGS ${INPUT_SNDFILE_CFLAGS})
   endif()
   if(INPUT_SNDFILE_LDFLAGS)
     to_space_list(INPUT_SNDFILE_LDFLAGS)
-    set_target_properties(input_sndfile PROPERTIES
+    set_target_properties(input_sndfile-freelcs  PROPERTIES
                LINK_FLAGS ${INPUT_SNDFILE_LDFLAGS})
   endif()
 endif()
diff -ur libebur128_fork_for_freelcs_3.4/scanner/inputaudio/sndfile/input_sndfile.c libebur128_fork_for_freelcs_3.4-patched/scanner/inputaudio/sndfile/input_sndfile.c
--- libebur128_fork_for_freelcs_3.4/scanner/inputaudio/sndfile/input_sndfile.c	2018-04-25 19:11:10.000000000 +0300
+++ libebur128_fork_for_freelcs_3.4-patched/scanner/inputaudio/sndfile/input_sndfile.c	2018-05-05 18:38:00.187483242 +0300
@@ -61,6 +61,7 @@
 }
 
 static int sndfile_set_channel_map(struct input_handle* ih, int* st) {
+  return 1;
   int result;
   int* channel_map = (int*) calloc((size_t) ih->file_info.channels, sizeof(int));
   if (!channel_map) return 1;
diff -ur libebur128_fork_for_freelcs_3.4/scanner/scanner.c libebur128_fork_for_freelcs_3.4-patched/scanner/scanner.c
--- libebur128_fork_for_freelcs_3.4/scanner/scanner.c	2018-04-25 19:11:10.000000000 +0300
+++ libebur128_fork_for_freelcs_3.4-patched/scanner/scanner.c	2018-05-05 18:38:00.187483242 +0300
@@ -82,6 +82,9 @@
     printf("  -m, --momentary=INTERVAL   print momentary loudness every INTERVAL seconds\n");
     printf("  -s, --shortterm=INTERVAL   print shortterm loudness every INTERVAL seconds\n");
     printf("  -i, --integrated=INTERVAL  print integrated loudness every INTERVAL seconds\n");
+    printf("\n");
+    printf("  Patched to disable progress bar.\n");
+    printf("\n");
 }
 
 static gboolean recursive = FALSE;
diff -ur libebur128_fork_for_freelcs_3.4/scanner/scanner-common/CMakeLists.txt libebur128_fork_for_freelcs_3.4-patched/scanner/scanner-common/CMakeLists.txt
--- libebur128_fork_for_freelcs_3.4/scanner/scanner-common/CMakeLists.txt	2018-04-25 19:11:10.000000000 +0300
+++ libebur128_fork_for_freelcs_3.4-patched/scanner/scanner-common/CMakeLists.txt	2018-05-05 18:38:00.187483242 +0300
@@ -2,7 +2,7 @@
 add_definitions(           ${GLIB20_CFLAGS_OTHER})
 
 add_library(scanner-common parse_args nproc scanner-common)
-target_link_libraries(scanner-common ebur128
+target_link_libraries(scanner-common ebur128-freelcs
                       ${GLIB20_LIBRARIES} ${GTHREAD20_LIBRARIES})
 
 if(SNDFILE_FOUND AND NOT DISABLE_SNDFILE)
diff -ur libebur128_fork_for_freelcs_3.4/scanner/scanner-common/scanner-common.c libebur128_fork_for_freelcs_3.4-patched/scanner/scanner-common/scanner-common.c
--- libebur128_fork_for_freelcs_3.4/scanner/scanner-common/scanner-common.c	2018-04-25 19:11:10.000000000 +0300
+++ libebur128_fork_for_freelcs_3.4-patched/scanner/scanner-common/scanner-common.c	2018-05-05 18:38:00.187483242 +0300
@@ -338,18 +338,21 @@
 
     // Start the progress bar thread. It misuses progress_mutex and
     // progress_cond to signal when it is ready.
-    g_mutex_lock(progress_mutex);
-    progress_bar_thread = g_thread_create(print_progress_bar,
-                                          &started, TRUE, NULL);
-    while (!started)
-        g_cond_wait(progress_cond, progress_mutex);
-    g_mutex_unlock(progress_mutex);
+    //
+    // Note progress bar causes hangs sometimes and this is why progress bar is disabled when using libebur128 with FreeLCS
+    //
+    // g_mutex_lock(progress_mutex);
+    // progress_bar_thread = g_thread_create(print_progress_bar,
+    //                                       &started, TRUE, NULL);
+    // while (!started)
+    //     g_cond_wait(progress_cond, progress_mutex);
+    // g_mutex_unlock(progress_mutex);
 
     pool = g_thread_pool_new((GFunc) init_state_and_scan_work_item,
                              opts, nproc(), FALSE, NULL);
     g_slist_foreach(files, (GFunc) init_state_and_scan, pool);
     g_thread_pool_free(pool, FALSE, TRUE);
-    g_thread_join(progress_bar_thread);
+    // g_thread_join(progress_bar_thread);
 }
 
 void print_version() {
diff -ur libebur128_fork_for_freelcs_3.4/test/CMakeLists.txt libebur128_fork_for_freelcs_3.4-patched/test/CMakeLists.txt
--- libebur128_fork_for_freelcs_3.4/test/CMakeLists.txt	2018-04-25 19:11:10.000000000 +0300
+++ libebur128_fork_for_freelcs_3.4-patched/test/CMakeLists.txt	2018-05-05 18:38:00.187483242 +0300
@@ -1,6 +1,6 @@
 if(SNDFILE_FOUND)
   include_directories(SYSTEM ${SNDFILE_INCLUDE_DIRS})
   add_executable(r128-test-library tests.c)
-  target_link_libraries(r128-test-library ebur128 ${SNDFILE_LIBRARIES})
+  target_link_libraries(r128-test-library ebur128-freelcs ${SNDFILE_LIBRARIES})
   add_test(test1 ${EXECUTABLE_OUTPUT_PATH}/r128-test-library)
 endif()
